plugins/optimization-detective/class-od-url-metric.php

```diff
@@ -77,39 +90,67 @@ private function validate_data( array $data ): void {
 	 * @return array<string, mixed> Schema.
 	 */
 	public static function get_json_schema(): array {
+		// See <https://developer.mozilla.org/en-US/docs/Web/API/DOMRectReadOnly>.
 		$dom_rect_schema = array(
 			'type'                 => 'object',
+			'required'             => true,
 			'properties'           => array(
 				'width'  => array(
-					'type'    => 'number',
-					'minimum' => 0,
+					'type'     => 'number', // An 'unrestricted double'.
+					'required' => true,
+					'minimum'  => 0.0,      // The spec allows this to be negative in DOMRectReadOnly, but this doesn't make sense in the context of intersectionRect and boundingClientRect.
 				),
 				'height' => array(
-					'type'    => 'number',
-					'minimum' => 0,
+					'type'     => 'number', // An 'unrestricted double'.
```

Comment:
why is this comment "an unrestricted double" and below L109 etc its "A double"? Are they different?

Reply:
Yeah, this is what it is referred to in the DOM spec: https://drafts.fxtf.org/geometry/#dom-domrectreadonly-width

This is probably too much information 😄

Reply:
Addressed in 7399131